### PR TITLE
Fix #4318 Default value not working

### LINKED ISSUE #4318: DefaultValue not working
Create EntityMetaData:

EditableEntityMetaData entityMetaData = new DefaultEntityMetaData("StringTest");
entityMetaData.addAttribute("col1").setDataType(STRING).setNillable(false).setIdAttribute(true);
entityMetaData.addAttribute("col2").setDataType(STRING);
entityMetaData.addAttribute("col3").setDataType(STRING).setDefaultValue("myDefault");

Create  repo:
dataService.getMeta().addEntityMeta(entityMetaData);

Add entity:
DefaultEntity entity = new DefaultEntity(entityMetaData, dataService);
entity.set("col1", "col1value");
entity.set("col2", "col2value");
dataService.add(entityMetaData.getName(), entity);

Retrieve entity:
Entity retrieved = dataService.findOne(entityMetaData.getName(), entity.getIdValue());

assertEquals(retrieved.get("col1"), "col1value");
assertEquals(retrieved.get("col2"), "col2value");
assertEquals(retrieved.get("col3"), "myDefault"); -> fails, is null

### PR DESCRIPTION
**Pull request by tommydeboer**
Fix #4318

### CODE REVIEW

File: molgenis-data/src/main/java/org/molgenis/data/support/EntityWithComputedAttributes.java [hunk 1]
```diff
@@ -28,6 +28,8 @@ public EntityWithComputedAttributes(Entity entity)
 				expressionEvaluators.put(amd.getName(), ExpressionEvaluatorFactory.createExpressionEvaluator(amd, emd));
 			}
 		}
+
+		// don't set default values, the decorated entity will already have set those
```
Comment (Mark-de-Haan) [inline molgenis-data/src/main/java/org/molgenis/data/support/EntityWithComputedAttributes.java]: Remove comments

File: molgenis-data/src/main/java/org/molgenis/data/settings/DefaultSettingsEntity.java [hunk 2]
```diff
@@ -28,6 +28,8 @@
 	public DefaultSettingsEntity(String entityId)
 	{
 		this.entityName = SettingsEntityMeta.PACKAGE_NAME + '_' + entityId;
+
+		// do not set default values, this is done at context refresh in DefaultSettingsEntityMetaData
```
Comment (Mark-de-Haan) [inline molgenis-data/src/main/java/org/molgenis/data/settings/DefaultSettingsEntity.java]: Remove comments

File: molgenis-data/src/main/java/org/molgenis/data/support/LazyEntity.java [hunk 3]
```diff
@@ -28,6 +28,8 @@ public LazyEntity(EntityMetaData entityMetaData, DataService dataService, Object
 		this.entityMetaData = requireNonNull(entityMetaData);
 		this.dataService = requireNonNull(dataService);
 		this.id = requireNonNull(id);
+
+		// don't set default values, the lazy loaded entity will already have set those
```
Comment (Mark-de-Haan) [inline molgenis-data/src/main/java/org/molgenis/data/support/LazyEntity.java]: Remove comments

File: molgenis-data/src/main/java/org/molgenis/data/support/PartialEntity.java [hunk 4]
```diff
@@ -28,6 +28,8 @@ public PartialEntity(Entity decoratedEntity, Fetch fetch, EntityManager entityMa
 		this.decoratedEntity = requireNonNull(decoratedEntity);
 		this.fetch = requireNonNull(fetch);
 		this.entityManager = requireNonNull(entityManager);
+
+		// don't set default values, the decorated entity will already have set those
```
Comment (Mark-de-Haan) [inline molgenis-data/src/main/java/org/molgenis/data/support/PartialEntity.java]: Remove comments

**Comment by Mark-de-Haan** [comment]
Feels like a big complicated fix.
Is it necessary for so many classes to do a `setDefaultValues(this) `?

**Comment by tommydeboer** [comment]
It's an entity's responsibility to set its default values at construction (see the interface). Since we have many entity implementations all those classes have to call ```setDefaultValues()```. 

Some entities that wrap other entities don't need to set default values. The comments are there to make that explicit and to avoid confusion.

**Pull request merged**